src/Strategy/BerkeleyDB.hs

```diff
@@ -0,0 +1,189 @@
+  pure $
+    DependencyResults
+      { dependencyGraph = graph
+      , dependencyGraphBreadth = graphBreadth
+      , dependencyManifestFiles = [dbFile project]
+      }
+
+-- | An entry in the database, consisting of the architecture, package, and version.
+data BdbEntry = BdbEntry
+  { bdbEntryArch :: Text
+  , bdbEntryPackage :: Text
+  , bdbEntryVersion :: Text
+  }
+  deriving (Eq, Ord, Show)
+
+-- | FOSSA _requires_ that architecture is provided: https://github.com/fossas/FOSSA/blob/e61713dec1ef80dc6b6114f79622c14df5278235/modules/fetchers/README.md#locators-for-linux-packages
+parsePkgInfo :: (Has Diagnostics sig m) => PkgInfo -> m BdbEntry
+parsePkgInfo PkgInfo{pkgArch = Just (pkgArch), ..} = pure $ BdbEntry pkgArch pkgName (pkgVersion <> "-" <> pkgRelease)
+parsePkgInfo pkg = fatalText . toText $ "package '" <> show pkg <> "' does not have an architecture, which is required"
+
+analyze ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has ReadFS sig m
+  , Has Exec sig m
+  ) =>
+  Path Abs Dir ->
+  Path Abs File ->
+  OsInfo ->
+  m (Graphing Dependency, GraphBreadth)
+analyze _ file osInfo = do
+  installed <- context ("read berkeleydb database file: " <> toText file) $ readBerkeleyDB file
+  context "building graph of packages" $ pure (buildGraph osInfo installed, Complete)
+
+buildGraph :: OsInfo -> [BdbEntry] -> Graphing Dependency
+buildGraph (OsInfo os osVersion) = directs . map toDependency
+  where
+    toDependency :: BdbEntry -> Dependency
+    toDependency pkg =
+      Dependency
+        LinuxRPM
+        (bdbEntryPackage pkg <> "#" <> os <> "#" <> osVersion)
+        (Just $ version pkg)
+        mempty
+        mempty
+        mempty
+
+    version :: BdbEntry -> VerConstraint
+    version pkg = CEq $ (bdbEntryArch pkg) <> "#" <> (bdbEntryVersion pkg)
+
+-- | Packages are read as a JSON array of base64 strings.
+readBerkeleyDB ::
+  ( Has Diagnostics sig m
+  , Has (Lift IO) sig m
+  , Has ReadFS sig m
+  , Has Exec sig m
+  ) =>
+  Path Abs File ->
+  m [BdbEntry]
+readBerkeleyDB file = withBerkeleyBinary $ \bin -> do
+  -- Get the process working directory, not the one that 'ReadFS' reports, because 'ReadFS' is inside of the tarball.
+  cwd <- getSystemCwd
+
+  -- Read the file content from disk and send it to the parser via stdin.
+  -- This is necessary because the parser doesn't have access to the file system being read.
+  fileContent <- context "read file content" $ readContentsBS file
+
+  -- Handle the JSON response.
+  (bdbJsonOutput :: [Text]) <- context "read raw blobs" . execJson' cwd (bdbCommand bin) . decodeUtf8 $ B64.encode fileContent
+  bdbByteOutput <- context "decode base64" . traverse fromEitherShow $ B64.decode <$> fmap encodeUtf8 bdbJsonOutput
+  entries <- context "parse blobs" . traverse fromEitherShow $ readPackageInfo <$> fmap BSL.fromStrict bdbByteOutput
+  context "parse package info" $ traverse parsePkgInfo entries
```

Comment:
We will need to account for epoch value. In backend it's not used to query knowledge api, but I think we persist locators with epoch value for rpm.  

```bash
✘ megh@Mac-mini  ~/test  fossa container analyze centos:7 --output | jq | grep "findutils" -C 20
        "metadata": {
          "architecture": "aarch64",
          "epoch": 1,
          "license": "GPLv3+",
          "name": "findutils",
          "release": "6.el7",
          "size": 1888914,
          "sourceRpm": "findutils-4.5.11-6.el7.src.rpm",
          "vendor": "CentOS",
          "version": "4.5.11"
        },
```

- https://github.com/knqyf263/go-rpmdb/blob/9a3bd2ebb923399db2189c5d8963af27368ba2c8/pkg/rpmtags.go#L16

Reply:
Nice catch. Fixed!